pkg/controller/atlascluster/atlascluster_controller.go

```diff
@@ -44,22 +46,20 @@ func (r *AtlasClusterReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error
 	log := r.Log.With("atlascluster", req.NamespacedName)
 
 	cluster := &mdbv1.AtlasCluster{}
-	if result := customresource.GetResource(r.Client, req, cluster, log); !result.IsOk() {
+	result, wctx := customresource.PrepareResource(r.Client, req, cluster, log)
+	if !result.IsOk() {
 		return result.ReconcileResult(), nil
 	}
 
-	log = log.With("clusterName", cluster.Spec.Name)
+	log.Infow("-> Starting AtlasCluster reconciliation", "spec", cluster.Spec)
+	defer statushandler.Update(wctx, r, cluster)
 
 	project := &mdbv1.AtlasProject{}
-	if result := customresource.GetResource(r.Client, req, project, log); !result.IsOk() {
+	if result := readProjectResource(r, cluster, project); !result.IsOk() {
```

Comment:
@vasilevp  this is the fix for reading the project - now `PrepareResource` doesn't fit any more as we just simply need to read the Custom resource